File: Task/Bernoullis-triangle/Java/bernoullis-triangle.java
import java.util.ArrayList;
import java.util.List;

public final class BernoullisTriangle {

	public static void main(String... args) {
		List<Integer> previousRow = new ArrayList<Integer>();
		
		for ( int n = 0; n < 15; n++ ) {
			List<Integer> currentRow = new ArrayList<Integer>();
		    for ( int k = 0; k <= n; k++ ) {
		        if ( k == 0 ) {
		            currentRow.addLast(1);
		        } else if ( k < n ) {
		            currentRow.addLast(previousRow.get(k) + previousRow.get(k - 1));
		        } else {
		            currentRow.add(2 * previousRow.getLast());
		        }
		    }
		
		    IO.println(currentRow);		
		    previousRow = currentRow;
		}
	}

}

File: Task/Bernoullis-triangle/Rust/bernoullis-triangle.rs
fn main() {
    let mut previous_row: Vec<u32> = Vec::new();
    for n in 0..15 {
        let mut current_row: Vec<u32> = Vec::new();
        for k in 0..=n {
            if k == 0 {
                current_row.push(1);
            } else if k < n {
                current_row.push(previous_row[k] + previous_row[k - 1]);
            } else {
                current_row.push(2 * previous_row[previous_row.len() - 1]);
            }
        }
        for entry in &current_row {
            print!("{:<6}", entry);
        }
        println!();
        previous_row = current_row;
    }
}

